src/ServicePulse.Host/vue/src/stores/CustomChecksStore.ts

```diff
@@ -0,0 +1,53 @@
+import { useDeleteFromServiceControl, useTypedFetchFromServiceControl } from "@/composables/serviceServiceControlUrls";
+import CustomCheck from "@/resources/CustomCheck";
+import { acceptHMRUpdate, defineStore } from "pinia";
+import { ref, watch } from "vue";
+import useAutoRefresh from "@/composables/autoRefresh";
+
+export const useCustomChecksStore = defineStore("CustomChecksStore", () => {
+  const prefix = "customchecks/";
+
+  const pageNumber = ref(1);
+  const failingCount = ref(0);
+  const failedChecks = ref<CustomCheck[]>([]);
+
+  const dataRetriever = useAutoRefresh(async () => {
+    try {
+      const [response, data] = await useTypedFetchFromServiceControl<CustomCheck[]>(`customchecks?status=fail&page=${pageNumber.value}`);
+      failedChecks.value = data;
+      failingCount.value = parseInt(response.headers.get("Total-Count") ?? "0");
+    } catch (e) {
+      failedChecks.value = [];
+      failingCount.value = 0;
+      throw e;
+    }
+  }, 5000);
+
+  watch(pageNumber, () => dataRetriever.executeAndResetTimer());
+
+  async function dismissCustomCheck(id: string) {
+    dataRetriever.executeAndResetTimer(async () => {
```

Comment:
probably want a comment here about the possibility of a zombie check appearing after next refresh if delete hasn't completed yet